From f741358022646f390154cfc922bece654df26741 Mon Sep 17 00:00:00 2001
From: mokopatches <mokopatches@openmoko.org>
Date: Fri, 4 Apr 2008 11:33:02 +0100
Subject: [PATCH] g_ether-vendor_product.patch
 Use FIC's own USB Vendor ID rather than NetChip's

Yes, we could solve this by some modprobe.conf parameters, but I'd like to
rather not rely on this.
---
 drivers/usb/gadget/ether.c |   11 ++++-------
 1 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/usb/gadget/ether.c b/drivers/usb/gadget/ether.c
index a7f38b6..6efbb44 100644
--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -134,11 +134,8 @@ struct eth_dev {
  * Instead:  allocate your own, using normal USB-IF procedures.
  */
 
-/* Thanks to NetChip Technologies for donating this product ID.
- * It's for devices with only CDC Ethernet configurations.
- */
-#define CDC_VENDOR_NUM	0x0525		/* NetChip */
-#define CDC_PRODUCT_NUM	0xa4a1		/* Linux-USB Ethernet Gadget */
+#define CDC_VENDOR_NUM	0x1457		/* First International Computer */
+#define CDC_PRODUCT_NUM	0x5117		/* Linux-USB Ethernet Gadget */
 
 /* For hardware that can't talk CDC, we use the same vendor ID that
  * ARM Linux has used for ethernet-over-usb, both with sa1100 and
@@ -159,8 +156,8 @@ struct eth_dev {
  * used with CDC Ethernet, Linux 2.4 hosts will need updates to choose
  * the non-RNDIS configuration.
  */
-#define RNDIS_VENDOR_NUM	0x0525	/* NetChip */
-#define RNDIS_PRODUCT_NUM	0xa4a2	/* Ethernet/RNDIS Gadget */
+#define RNDIS_VENDOR_NUM	0x1457	/* NetChip */
+#define RNDIS_PRODUCT_NUM	0x5122	/* Ethernet/RNDIS Gadget */
 
 
 /* Some systems will want different product identifers published in the
-- 
1.5.6.5

